cosmwasm/packages/wasmi-runtime/src/crypto/multisig.rs

```diff
@@ -152,3 +151,113 @@ fn decode_multisig_signature(raw_blob: &[u8]) -> Result<MultisigSignature, Crypt
+    use crate::crypto::multisig::{decode_multisig_signature, MultisigSignature};
+
+    pub fn test_decode_sig_sanity() {
+        let sig: Vec<u8> = vec![
+            0, 0, 0, 0, 0, 0, 0, 0x12, 10, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 0x12, 4, 1, 2, 3, 4,
+        ];
+
+        let result = decode_multisig_signature(sig.as_slice());
+        assert!(result.is_ok());
+
+        let result = result.unwrap();
+        assert_eq!(
+            result,
+            vec![vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10], vec![1, 2, 3, 4]],
+            "Signature is: {:?} and result is: {:?}",
+            sig,
+            result
+        )
+    }
+
+    pub fn test_decode_long_leb128() {
+        let sig: Vec<u8> = vec![
+            0, 0, 0, 0, 0, 0, 0, 0x12, 200, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0,
+        ];
+
+        let result = decode_multisig_signature(sig.as_slice());
+        assert!(result.is_ok());
+
+        let result = result.unwrap();
+        assert_eq!(
+            result,
+            vec![vec![
+                0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0,
+            ]],
+            "Signature is: {:?} and result is: {:?}",
+            sig,
+            result
+        )
+    }
+
+    pub fn test_decode_wrong_long_leb128() {
+        let malformed_sig: Vec<u8> = vec![
+            0, 0, 0, 0, 0, 0, 0, 0x12, 205, 1, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15,
+        ];
+
+        let result = decode_multisig_signature(malformed_sig.as_slice());
+        assert!(
+            result.is_err(),
+            "Signature is: {:?} and result is: {:?}",
+            malformed_sig,
+            result
+        );
```

Comment:
you can probably assert equality with a specific error, if the error type implements `PartialEq` and either `Debug` or `Display` (not sure which one). you can import a quick and dirty `Display` derive macro from `derive_more::Display` which is available

Reply:
Not sure that it makes a difference lol

Reply:
besides, all errors there are `CryptoError::ParsingError` so..

Reply:
The main difference is in the error message if it fails, but yeah this isn't really important in this case.